sensor/common/image/service_impl.go

```diff
@@ -67,16 +63,6 @@ func (s *serviceImpl) SetClient(conn grpc.ClientConnInterface) {
 }
 
 func (s *serviceImpl) GetImage(ctx context.Context, req *sensor.GetImageRequest) (*sensor.GetImageResponse, error) {
-	if id := req.GetImage().GetId(); id != "" {
-		v, _ := s.imageCache.Get(imagecacheutils.GetImageCacheKey(req.GetImage()))
-		img, _ := v.(*storage.Image)
```

Comment:
What was the problem with the fix you did in your other PR?
```golang
	id := req.GetImage().GetId()
	v, ok := s.imageCache.Get(imagecacheutils.GetImageCacheKey(req.GetImage()))
	if id != "" && ok {
		img := v.WaitAndGet()
```

Reply:
Tests are failing :(
I can spent some time to debug it but I have not much context why maybe they wait forever :thinking: 
Maybe we need another cache that will just store images without this wait logic

Reply:
I think that having another cache with only the images could solve the issue but we will have to sync the two caches and that could lead to other complications. I think it's worth trying to understand the problem and fix it. Which tests in particular are failing because of this? I saw a lot of tests failing in the other PR, but I'm not sure which ones are related to this.

Interestingly, I found something that, while I think it works, it might work by accident 😆 . [Here](https://github.com/stackrox/stackrox/blob/7d64f4eb4e1eecac3fcd8f09bbd31266d0b12910/sensor/common/detector/detector.go#L329-L340) we create a new `cacheValue` but we do not create the signal, nor do we trigger it. I think this works because the default state of a `Signal` is 'triggered' so that `cacheValue` will not block if `WaitAndGet` is called, but I think this is by accident and not really intentional.